components/compositing/webview_renderer.rs

```diff
@@ -807,13 +807,15 @@ impl WebViewRenderer {
 
         // Batch up all scroll events into one, or else we'll do way too much painting.
         let mut combined_scroll_event: Option<ScrollEvent> = None;
-        let mut combined_magnification = 1.0;
+        let mut combined_magnification = self.pinch_zoom_level().get();
         for scroll_event in self.pending_scroll_zoom_events.drain(..) {
             match scroll_event {
-                ScrollZoomEvent::PinchZoom(magnification) |
-                ScrollZoomEvent::ViewportZoom(magnification) => {
+                ScrollZoomEvent::PinchZoom(magnification) => {
                     combined_magnification *= magnification
                 },
+                ScrollZoomEvent::InitialViewportZoom(magnification) => {
+                    combined_magnification = magnification
+                },
                 ScrollZoomEvent::Scroll(scroll_event_info) => {
```

Comment:
This will have the wrong behavior if pinch zooming while a page loads. Perhaps something like this makes sense:

    let mut base_page_zoom = self.pinch_zoom_level.get();
    let mut combined_page_zoom_modification = 1.0;
    for scroll_event in self.pending_scroll_zoom_events.drain(..) {
        match scroll_event {
            ScrollZoomEvent::PinchZoom(magnification) => {
                combined_page_zoom_modication *= magnification
             },
             ScrollZoomEvent::InitialViewportZoom(magnification) => {
                 base_page_zoom = magnification
             },
...
```